### ISSUE TITLE
Toolbar elements not accessible to screen readers

### ISSUE THREAD

**Issue by Ksan8** [quilljs/quill#3320]
Any app that uses Quill will fail accessibility checks (such as the [axe browser extension](https://www.deque.com/axe/browser-extensions/)) due to the toolbar not being accessible to screen readers. This is a critical accessibility issue.

The issues Axe finds are:
- ARIA commands must have an accessible name
- Buttons must have discernible text

Providing a [minimal, complete and verifiable example](https://stackoverflow.com/help/mcve) will further increase your chances that someone can help.

**Steps for Reproduction**

1. Visit any webpage or app that uses the Quill editor (with at least one dropdown and a button), or download [quill-minimal-example](https://github.com/Ksan8/quill-minimal-example) and follow the instructions to start a dev server
2. Install the [axe browser extension](https://www.deque.com/axe/browser-extensions/) if not already installed
3. Open DevTools and click on the axe tab
4. Choose "Scan all of my page" after ensuring that all content is within view
5. Notice the issues listed above appear in the axe results

**Expected behavior**:
All toolbar dropdowns and buttons should have `aria-label` attributes, which would allow them to be discernible to screen readers.

**Actual behavior**:
Currently, an element (such as a dropdown or picker) with `.ql-picker-label` have an `aria-controls` attribute, but no `aria-label`. They also have no `title` attribute, or other text that is visible to screen readers. All other buttons within the toolbar lack `aria-label` or `title` attributes as well. 

**Platforms**:
- Brave version 1.21.76 and Chrome version 89.0.4389.90
- MacOS Big Sur version 11.2.3

**Version**:
- Quill version 1.3.7
- ngx-quill version 13.2.0

**Comment by clairefields15** [comment]
I can work on this, having the same issue using this editor at work and seems like a simple fix.

**Comment by jobara** [comment]
It looks like for the buttons you might just need to add the `format` as an `aria-label` to the [`addButton`](https://github.com/quilljs/quill/blob/541bd363ac8eae2ad876f16a950cea439eae262a/modules/toolbar.js#L153-L161) method. You might also need the `value` for things like lists. That was just for a quick look though, I'm not sure what's needed for the select boxes and other things.

For those looking for a temporary workaround, it may be possible to make a custom toolbar that included the aria-labels. See: https://quilljs.com/docs/modules/toolbar/

**Comment by clairefields15** [comment]
I have all the changes necessary to implement this much needed accessibility work done locally and tested, but I cannot push to the repo to make a PR, can someone help me out?
`ERROR: Permission to quilljs/quill.git denied to clairefields15.
fatal: Could not read from remote repository. Please make sure you have the correct access rights
and the repository exists.`

@jobara unfortunately the temporary workaround does not work, check out [this code sandbox](https://codesandbox.io/s/6x93pk4rp3) using a screenreader. It looks like quill just looks for matching `ql-foo` containers and then inserts it's own markup (sans aria-labels) into those containers.

**Comment by hchang21** [comment]
@clairefields15 Any updates on this issue? I'm using v1.3.7, and the only part of the toolbar that reads with a screen reader is the option to set a heading level. Everything else still reads as blank buttons.

**Comment by samuelClo** [comment]
> @clairefields15 Any updates on this issue? I'm using v1.3.7, and the only part of the toolbar that reads with a screen reader is the option to set a heading level. Everything else still reads as blank buttons.

You can make your [own toolbar html](https://quilljs.com/docs/modules/toolbar/#:~:text=For%20use%20cases%20requiring%20even%20more%20customization%2C%20you%20can%20manually%20create%20a%20toolbar%20in%20HTML%2C%20and%20pass%20the%20DOM%20element%20or%20selector%20into%20Quill.) and add aria label.

But it will be good to have a native implementation of basic accessibility.

**Comment by hchang21** [comment]
Thanks for the tip
